src/ripple/nodestore/impl/DatabaseShardImp.h

```diff
@@ -297,6 +326,38 @@ class DatabaseShardImp : public DatabaseShard
 
     void
     removeFailedShard(std::shared_ptr<Shard> shard);
+
+    // Returns the index that represents the logical
+    // partition between historical and recent shards
+    std::uint32_t
+    getShardBoundaryIndex(std::lock_guard<std::mutex> const&);
+
+    std::uint32_t
+    countHistoricalShards(std::lock_guard<std::mutex> const&);
+
+    // Shifts the recent and second most recent (by index)
+    // shards as new shards become available on the network.
+    // Older shards are moved to a historical shard path.
+    void
+    checkRecentShards(std::lock_guard<std::mutex> const& lock);
```

Comment:
I think that's misleading as well since the majority of calls to this function will not result in any relocation.

Reply:
Changed to `relocateOutdatedShards`

Reply:
Yep, that's a better name.

Reply:
The new `relocateOutdatedShards` name is better, since it makes it clear _which_ shards will be relocated, but just FYI another convention I've seen is "maybe", so `maybeRelocateShards`.

Reply:
That's funny I actually considered "maybe" but wasn't sure if that would be accepted.